src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -385,10 +383,299 @@ public static void EnsureSocketsAreInitialized()
                             throw new SocketException((int)errorCode);
                         }
 
+                        s_getAddrInfoExSupported = TestGetAddrInfoEx();
+
                         Volatile.Write(ref s_initialized, true);
                     }
                 }
             }
         }
+
+        private static bool TestGetAddrInfoEx()
+        {
+            using (SafeLibraryHandle libHandle = Interop.Kernel32.LoadLibraryExW(Interop.Libraries.Ws2_32, IntPtr.Zero, 0))
+            {
+                return Interop.Kernel32.GetProcAddress(libHandle, nameof(Interop.Winsock.GetAddrInfoExCancel)) != IntPtr.Zero;
```

Comment:
We're testing for GetAddrInfoExCancel because that serves as an indication for whether GetAddrInfoExW supports overlapped execution?  Can you add a comment?

Reply:
Is there something we can assert to help validate that this is working? e.g. checking the Windows version and asserting that TestGetAddrInfoEx returns true on an appropriately recent version?